tests/Unit/IO/H5/Python/Test_ExtractDatFromH5.py

```diff
@@ -60,15 +60,22 @@ def test_extract_dat_files(self):
 
         # Test that we get an error if we try to run again and 'force' flag is
         # False.
-        self.failUnlessRaises(ValueError,
-                              extract_dat_files,
-                              self.h5_filename,
-                              out_dir=self.created_dir)
+        self.failUnlessRaises(ValueError, extract_dat_files, self.h5_filename,
+                              self.created_dir, 1)
 
         # Try with 'force' flag True this time
-        extract_dat_files(self.h5_filename,
-                          out_dir=self.created_dir,
-                          force=True)
+        extract_dat_files(self.h5_filename, self.created_dir, 1, force=True)
+
+        memory_data = np.loadtxt(memory_data_path)
+        timestep_data = np.loadtxt(timestep_data_path)
+
+        self.assertEqual(memory_data.all(), expected_memory_data.all())
+        self.assertEqual(timestep_data.all(), expected_timestep_data.all())
+
+        self.tearDown()
```

Comment:
Don't call tearDown yourself (is that safe?). Just split into a separate function if you need to.